shared/cluster.go

```diff
@@ -633,3 +633,64 @@ func FetchToken(ip string) (string, error) {
 
 	return token, nil
 }
+
+func CreateSecret(secret, namespace string) error {
+	kubectl := fmt.Sprintf("kubectl --kubeconfig %s", KubeConfigFile)
+	if namespace == "" {
+		namespace = "default"
+	}
+	cmd := fmt.Sprintf("%s create secret generic %s -n %s --from-literal=mykey=mydata",
+		kubectl, secret, namespace)
+	createStdOut, err := RunCommandHost(cmd)
+	if err != nil {
+		return ReturnLogError("failed to create secret: \n%w", err)
+	}
+	if strings.Contains(createStdOut, "failed to create secret") {
+		return ReturnLogError("failed to create secret: \n%w", err)
+	}
+	LogLevel("DEBUG", "Create Secret Output: %s", createStdOut)
```

Comment:
so this is a public function, which means to be consumed.

lets talk about it on monday with everyone, but we should try to only log on the provider if its something very occasional and unexpected or a complicated flow.

so if i am a consumer of this function, and dont want or cant have a log from my secret, wherever the log will be output, i have no option, if i consume this func, mandatory i will have this log .

Reply:
added a print bool - to give the option of printing vs not for now. The day we can actually use the implementation for "DEBUG" like its supposed to, this will hopefully not be a problem anymore.

Reply:
so you mentioned somewhere else ( which i totally agree ) that we in the future will have a "start option" mode to differentiate what to log or not

we should deal that on that implementation, not adding arbitrary ifs boolean in the code 

also adding the print does not solve the issue i mentioned , its just a workaround that would need refactor later on 

 + you are adding unnecessary signature complexity in favor of a just workaround only

Reply:
fixed